yb-voyager/src/queryissue/object_collector.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright (c) YugabyteDB, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package queryissue
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/samber/lo"
+	log "github.com/sirupsen/logrus"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/queryparser"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+// ObjectCollector collects unique schema-qualified object names.
+type ObjectCollector struct {
```

Comment:
Nothing obviously wrong with the current approach, but how about: 
- Convert `TraverseParseTree` to a struct,
- Collect objects by default while traversing. Let that logic live inside `TraverseParseTree` struct instead of `ObjectCollector` struct. 

Reason being, object collection is core to the traversal, it has nothing to do with detecting issues, so I think it can live inside `queryparser` pkg and be part of the traversal logic directly. (instead of being in the queryissue pkg)
In `queryissue` pkg, your code remains the same, but instead of getting objects, schemas from objectCollector, you get it directly from the traversal struct.

What do you think?

Reply:
ObjectCollector can remain a separate struct if that makes sense, but let `Traversal` struct expose the methods to get the objects/schemas, and have `queryissue` deal directly with traversal struct, not objectCollector struct.

Reply:
hmm let me think about this. Moving the collection logic to queryparser packge makes sense, since these are low level details

Reply:
done